haystack/document_stores/filter_utils.py

```diff
@@ -0,0 +1,280 @@
+from typing import Union, List, Dict
+from abc import ABC, abstractmethod
+from collections import defaultdict
+
+
+def nested_defaultdict():
```

Comment:
Pls add a docstring why this method is needed (I think I saw the same in elasticsearch.py, too)

Reply:
Done. Forgot to remove the one from `elasticsearch.py`, so removed it now.